File: app/src/main/java/com/softaai/designpatterns/java/structural/proxy/RealImage.java
package com.softaai.designpatterns.java.structural.proxy;

/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
public class RealImage implements Image {
    private String fileName;

    public RealImage(String fileName){
        this.fileName = fileName;
        loadFromDisk(fileName);
    }
    @Override
    public void display() {
       System.out.println("Displaying " + fileName);
    }

    private void loadFromDisk(String fileName){
        System.out.println("Loading " + fileName);
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/Proxy/RealImage.kt
package com.softaai.designpatterns.kotlin.structural.Proxy


/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
class RealImage(fileName : String) : Image {
    private lateinit var fileName : String

    init {
        this.fileName = fileName
        loadFromDisk(fileName)
    }

    override fun display() {
        println("Displaying " + fileName)
    }

    fun loadFromDisk(fileName: String){
        println("Loading " + fileName)
    }
}